src/home/NotificationCenter.tsx

```diff
@@ -134,6 +141,12 @@ export default function Notifications({ navigation }: NotificationsProps) {
 
   const notifications = useNotifications()
 
+  // Changing onViewableItemsChanged on the fly is not supported.
+  // This is a workaround to provide handleViewableItemsChanged with
+  // actual notifications array while keeping its dependecy list empty.
+  const notificationsRef = useRef<Notification[]>([])
+  notificationsRef.current = notifications
```

Comment:
maybe we should have a `useEffect` here that updates the ref when `notifications` changes? i think notifications can potentially change while the user is on the screen because the app talks to firebase rtdb directly

Reply:
That way we save the `notifications` value in a ref every time a component re-renders. It does that in particular when rtdb will send an update and the `useNotifications()` hook will be triggered with a new value for `notifications`. 

Wrapping ref assignment into a `useEffect` will scope ref update only to re-renders caused by `notifications` change. Looks like that way it makes the initial intent more clear, I will refactor it to `useEffect`. 

Thanks for the suggestion!